packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -315,26 +315,25 @@ def _validate_overlay_subnet(agent_subnet, overlay_subnet, prefixlen):
 def _validate_overlay_backend(overlay_name, backend):
     try:
         # Make sure the backend has the right VNI.
-        vxlan = backend.pop('vxlan')
+        vxlan = backend['vxlan']
```

Comment:
🤔 Wonder why used `backend.pop('vxlan')` ... I can't find any place where we seem to rely on this test mutating the dictionary but feel uncomfortable green-lighting the PR as is without further investigation esp. as we seem to access quite some elements using `pop` in this file.

Reply:
`pop` was validating two things:
1. The object that we want to validate
2. The overall structure of the backend

The change has taken out the second test. And, so if there are some extra key-value pairs in the backend json, it will go unvalidated. Maybe, it was a good thing to validate the backend structure as well.

Reply:
I wonder if it would be possible to validate it without mutating the `backend`? Asserting something on the array of keys maybe?

Reply:
@GoelDeepak I wasn't referring to the `vxlan.pop('vtep_*')` but `backend.pop('vxlan')` call. I understand that we were removing the `vtep_*`  elements and then comparing the rest to an `expected_vxlan` dictionary verifying that only the specified elements were present and that we're now no longer failing the validation in case there are _unknown_ elements which I would argue is ok. I'm also very much in support of dropping the `pop` calls as I'd argue that they make it uncessary hard to follow the code.

I've further reviewed the flow and couldn't spot any place where we're relying on backend being mutated.